src/AssertRequestAgainstSchema.php

```diff
@@ -0,0 +1,121 @@
+use ByJG\ApiTools\Exception\StatusCodeNotMatchedException;
+
+trait AssertRequestAgainstSchema
+{
+    /**
+     * @var Schema
+     */
+    protected $schema;
+
+    /**
+     * configure the schema to use for requests
+     *
+     * When set, all requests without an own schema use this one instead.
+     *
+     * @param Schema|null $schema
+     */
+    public function setSchema($schema)
+    {
+        $this->schema = $schema;
+    }
+
+    /**
+     * @param string $method The HTTP Method: GET, PUT, DELETE, POST, etc
+     * @param string $path The REST path call
+     * @param int $statusExpected
+     * @param array|null $query
+     * @param array|null $requestBody
+     * @param array $requestHeader
+     * @return mixed
+     * @throws DefinitionNotFoundException
+     * @throws GenericSwaggerException
+     * @throws HttpMethodNotFoundException
+     * @throws InvalidDefinitionException
+     * @throws NotMatchedException
+     * @throws PathNotFoundException
+     * @throws StatusCodeNotMatchedException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @deprecated Use assertRequest instead
+     */
+    protected function makeRequest(
+        $method,
+        $path,
+        $statusExpected = 200,
+        $query = null,
+        $requestBody = null,
+        $requestHeader = []
+    ) {
+        $this->checkSchema();
+        $requester = new ApiRequester();
+        $body = $requester
+            ->withSchema($this->schema)
+            ->withMethod($method)
+            ->withPath($path)
+            ->withQuery($query)
+            ->withRequestBody($requestBody)
+            ->withRequestHeader($requestHeader)
+            ->assertResponseCode($statusExpected)
+            ->send();
+
+        // Note:
+        // This code is only reached if the send is successful and
+        // all matches are satisfied. Otherwise an error is throwed before
+        // reach this
+        $this->assertTrue(true);
+
+        return $body;
+    }
+
+    /**
+     * @param AbstractRequester $request
+     * @return mixed
+     * @throws DefinitionNotFoundException
+     * @throws GenericSwaggerException
+     * @throws HttpMethodNotFoundException
+     * @throws InvalidDefinitionException
+     * @throws NotMatchedException
+     * @throws PathNotFoundException
+     * @throws StatusCodeNotMatchedException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function assertRequest(AbstractRequester $request)
+    {
+        // Add own schema if nothing is passed.
+        if (!$request->hasSchema()) {
+            $this->checkSchema();
+            $request->withSchema($this->schema);
+        }
+
+        // Request based on the Swagger Request definitios
+        $body = $request->send();
+
+        // Note:
+        // This code is only reached if the send is successful and
+        // all matches are satisfied. Otherwise an error is throwed before
+        // reach this
+        $this->assertTrue(true);
```

Comment:
This is a wrong usage of a trait and typical mistake: this method is not defined in this trait and therefore it can not be used here. You have created a dependency which is not protected and leads to mistakes.

This will not work:

```php
class Example
{
   use \ByJG\ApiTools\AssertRequestAgainstSchema;

   public function foo()
   {
       $schema = \ByJG\ApiTools\Base\Schema::getInstance(
           file_get_contents('/path/to/json/definition')
       );
       $this->setSchema($schema);
       
       $request = new \ByJG\ApiTools\ApiRequester();
       $request
           ->withMethod('GET')
           ->withPath("/path/for/get/1");

       $this->assertRequest($request);
   }
}
```

Reply:
It's not _that_ bad. All that is missing is a line `assert($this instanceof TestCase);` at the beginning, plus a comment explaining that this is intended as plugin method for use in PHPUnit.

Reply:
Or, another method could be to add an abstract declaration of `assertTrue`, though that might be problematic to do for the different PHP and PHPUnit versions. I'm not sure if this even works technically though, that a trait defines an abstract function that is then implemented by the baseclass of the class using the trait.

Reply:
> It's not _that_ bad.

This has nothing to do with good or bad, it is wrong usage of a trait.

> I'm not sure if this even works technically though, that a trait defines an abstract function that is then implemented by the baseclass of the class using the trait.

https://www.php.net/manual/en/language.oop5.traits.php#language.oop5.traits.abstract

Reply:
Hi, not sure what you want me to do. 

1. Comment that the method is expected
2. add `abstract public static function assertTrue($condition, string $message = ''): void` -- could cause problem with php version + phpunit as @UlrichEckhardt states.
3. Add check `assert($this instanceof TestCase);`

Reply:
Of course, Trait will only work if Assert class is used.

Reply:
Last change of `assertTrue` was 3 years ago with initial implementation :)

Reply:
> Of course, Trait will only work if Assert class is used.

Exactly that is the misuse of a trait and `assert($this instanceof TestCase)` is only a workaround which doesn't make things any better.

> 2\. add `abstract public static function assertTrue($condition, string $message = ''): void` -- could cause problem with php version + phpunit as @UlrichEckhardt states.

Only with _dead_ versions of PHP and PHPUnit. Bump the PHP version is an option here but this pull request is already mixing up too many topics. _(imo)_

Reply:
Hi @froschdesign sorry for late response, lately I've noticed, that we do not need a the instance for assertTrue, we can use Assert::assertTrue! This will solve the not ideal "assertTrue" requirement. What do you think?

Reply:
Sorry, I am late to the party, but I have to agree that refer a suppose parent class inside a trait it is not the best approach. However if `Assert::assertTrue()` does not need any instance, I believe you solve the issue.